zrml/prediction-markets/src/lib.rs

```diff
@@ -135,21 +146,45 @@ mod pallet {
             origin: OriginFor<T>,
             market_id: MarketIdOf<T>,
         ) -> DispatchResultWithPostInfo {
-            T::DestroyOrigin::ensure_origin(origin)?;
+            ensure_root(origin)?;
 
             let mut total_accounts = 0usize;
             let mut share_accounts = 0usize;
             let market = T::MarketCommons::market(&market_id)?;
             let market_status = market.status;
             let outcome_assets = Self::outcome_assets(market_id, &market);
             let outcome_assets_amount = outcome_assets.len();
+
+            // Slash outstanding bonds
+            if market_status == MarketStatus::Proposed {
+                CurrencyOf::<T>::slash_reserved_named(
+                    &RESERVE_ID,
+                    &market.creator,
+                    T::AdvisoryBond::get(),
+                );
+            }
+            if market_status != MarketStatus::Resolved {
+                if market.creation == MarketCreation::Permissionless {
+                    CurrencyOf::<T>::slash_reserved_named(
+                        &RESERVE_ID,
+                        &market.creator,
+                        T::ValidityBond::get(),
+                    );
+                }
+                CurrencyOf::<T>::slash_reserved_named(
```

Comment:
Correct me if I am wrong, but the OracleBond is slashed in every case (excluding `resolved`). 

True, most of the time the OracleBond is still missing to slash, according to the table.

Take for example the `* -> InsufficientSubsidy` state change. The OracleBond is already slashed (`-O` in table) and you slash it again. Two is better, yes, but is this wanted? 

 Why do you excluding the `resolved` market status? I can't read that in the table (maybe `Reported/Disputed -> Resolved` but it makes no sense).

Reply:
I'm excluding `MarketStatus::Resolved` because the oracle bond is unreserved when the market is resolved, so we shouldn't slash that.

Good catch regarding `InsufficientSubsidy`. The function `process_subsidy_collecting_markets` handles unreserving of bonds, so there's no need to slash them in `destroy_market`. I'll submit a fix and check how this slipped through testing.